### PR TITLE
add class Machine 

### PR DESCRIPTION
**Pull request by felix5572**
add class Machine

### CODE REVIEW

File: dpdispatcher/batch_object.py [hunk 1]
```diff
@@ -9,6 +9,9 @@
 from dpdispatcher.ssh_context import SSHContext
 from dpdispatcher.submission import Resources
 
+# from typing import TypedDict
+from typing_extensions import TypedDict
```
Comment (njzjz) [inline dpdispatcher/batch_object.py:13]: this is a third-party module so you need to add it to setup.py

**Review by njzjz — CHANGES_REQUESTED** [review]

File: dpdispatcher/batch_object.py (hunk 1 above)
Comment (felix5572) [reply dpdispatcher/batch_object.py:13]: added

Comment (felix5572) [reply dpdispatcher/batch_object.py:13]: It looks like I use syntax supported only in python 3.9.
I will fix it

File: dpdispatcher/__init__.py [hunk 2]
```diff
@@ -8,7 +8,8 @@
 dlog = logging.getLogger(__name__)
 dlog.setLevel(logging.INFO)
 # dlogf = logging.FileHandler(os.getcwd()+os.sep+SHORT_CMD+'.log')
-dlogf = logging.FileHandler('./'+os.sep+SHORT_CMD+'.log')
+# dlogf = logging.FileHandler('./'+os.sep+SHORT_CMD+'.log')
+dlogf = logging.FileHandler(os.path.join(os.environ['HOME'], SHORT_CMD+'.log'))
```
Comment (njzjz) [inline dpdispatcher/__init__.py:12]: `os.path.expanduser('~')`

https://docs.python.org/3/library/os.path.html#os.path.expanduser

Comment (njzjz) [reply dpdispatcher/__init__.py:12]: But why did you change `dlogf`?

Comment (felix5572) [reply dpdispatcher/__init__.py:12]: Somestime, dpdispatcher may be used by other software (for me, airflow)
And './' may means '/' , and we can't write things there.

**Review by njzjz — APPROVED** [review]

**Pull request merged**